File: 2019-2020/pset1/mario.c
/**
 * mario.c
 *
 * Computer Science 50
 * Problem Set 1
 *
 * Prints out a half-pyramid of a specified height
 *
 * 2019:
 * https://lab.cs50.io/cs50/labs/2019/x/mario/less/
 * 2020:
 * https://lab.cs50.io/cs50/labs/2020/x/mario/less/
 *
 */

#include <stdio.h>
#include <cs50.h>

const char SHARP = '#';
const char BLANC = ' ';
const int MIN_HEIGHT = 1;
const int MAX_HEIGHT = 8;

void chart(char c, int n)
{
    for (int i = 0; i < n; ++i)
    {
        putchar(c);
    }
}

int main(void)
{
    int n;

    do
    {
        n = get_int("Height: ");
    }
    while ((n < MIN_HEIGHT) || (n > MAX_HEIGHT));

    for (int i = 1; i <= n; ++i)
    {
        chart(BLANC, n - i);
        chart(SHARP, i);
        printf("\n");        
    }
    return 0;
}

File: _time-worn/2019-2020/pset6/mario.py
# ------------------------------------------------------------------------------
# mario.py
# "Less comfortable" version
#
# Computer Science 50
# Problem Set 6
#
# Prints out a half-pyramid of a specified heigh
#
# 2020:
# https://cs50.harvard.edu/x/2020/psets/6/mario/less/
# 2019
# https://docs.cs50.net/2019/x/psets/6/sentimental/mario/less/mario.html
# ------------------------------------------------------------------------------

from cs50 import get_int

while True:
    height = get_int("Height: ")
    if 0 < height < 9:
        break

for k in range(height):
    print('{0}{1}  {1}'.format((height - k - 1) * ' ', (k + 1) * '#'))
